src/init.cpp

```diff
@@ -1561,7 +1561,7 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
 
                         const CBlockIndex* tip = chainstate->m_chain.Tip();
                         RPCNotifyBlockChange(tip);
-                        if (tip && tip->nTime > GetAdjustedTime() + 2 * 60 * 60) {
+                        if (tip && tip->nTime > GetTime() + MAX_FUTURE_BLOCK_TIME) {
```

Comment:
Should we have some extra tolerance here in case the adjusted time had been forward a bit?

Reply:
This pull request is a refactor and I recommend to do any behavior changes in separate pull requests.

Reply:
I don't understand this: This PR argues that `GetAdjustedTime()` can't differ from `GetTime()` at this point in init, so I can't see how it could have been "forward a bit". Or do you mean the former adjusted time at the point in time when we validated the tip?